packages/router-core/src/router.ts

```diff
@@ -2962,6 +3033,33 @@ export function getInitialRouterState(
     statusCode: 200,
   }
 }
+function validateState(validateState: AnyValidator, input: unknown): unknown {
```

Comment:
same here, can we move the validation of both state and search into a reusable function?